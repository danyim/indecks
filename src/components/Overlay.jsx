import React from 'react';
import styles from '../styles/components/Overlay.styl';

const propTypes = {
  className: React.PropTypes.string,
  children: React.PropTypes.node
};

const defaultProps = {
  className: '',
  children: null
};

const Overlay = (props) => {
  return (
    <div className={`${styles.overlay} ${props.className}`}>
      <div className={`${styles['hover-actions-container']}`}>
        {props.children}
      </div>
    </div>
  )
};

Overlay.propTypes = propTypes;
Overlay.defaultProps = defaultProps;

export default Overlay;
