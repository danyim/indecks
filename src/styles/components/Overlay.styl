@import '../_variables';

.overlay
  position: absolute
  top: 0
  left: 0
  height: 100%
  width: 100%
  // transition all 0.2s
  // transition-delay: 0.1s // Not sure if this is good or not, try A/Bing
  opacity: 0
  &:hover
    opacity: 1

.hover-actions-container
  height: 100%
  align-items: center
  justify-content: center

.hover-actions
  flex-direction: row
  opacity: 1
  color: red

.hover-button
  display: inline-block
  background-color: white
  box-shadow: 1 1 0 0 black
  padding: 1rem 2rem
  margin-left: 1rem
  :nth-child(-n+1)
    margin-left: 0
