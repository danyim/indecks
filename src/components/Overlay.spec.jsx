import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Overlay from './Overlay'

const defaultProps = {
  className: '',
  children: null
}

function setup(props = defaultProps) {
  const wrapper = shallow(<Overlay {...props} />)

  return {
    props,
    wrapper
  }
}

describe('Overlay', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup()
    expect(wrapper.find('div.overlay').exists()).toBe(true)
    expect(wrapper.find('div.hover-actions-container').exists()).toBe(true)

    const tree = renderer.create(<Overlay {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
