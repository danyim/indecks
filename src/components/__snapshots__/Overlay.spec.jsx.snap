// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render self and subcomponents 1`] = `
<div
  className="overlay "
>
  <div
    className="hover-actions-container"
  />
</div>
`
